yt

```diff
@@ -1,69 +1,191 @@
 #!/bin/bash
-# dependencies: mpv youtube-dl fzf rofi/dmenu
-# search videos and playlists on youtube and play them in mpv, without an API 
-# usage:
-# yt					asks for input in stdin, prompts using fzf
-# yt search query		takes input from the passed arg, prompts using fzf
-# yt -r					takes input and prompts using rofi ($guicmd)
+
+# explain usage
+function usage () {
+    echo "usage: $(basename $0)"
+    echo "    -h help"
+    echo "    -c channels/subscriptions"
+    echo "    -s search"
+    echo "    -g gui option"
+    echo "    nothing use defaults"
+    exit 1
+}
+
+# dont use defaults
+useDefaults="f"
+
+# no args -> use defaults
+if [[ ${#} -eq 0 ]]; then
+    useDefaults="t"
+fi
+
+# available flags
+optstring=":s:c:gh"
 
 defcmd="fzf"
-guicmd="rofi -dmenu -i" #uncomment next line for dmenu
-#guicmd="dmenu -i -l 15"
+defaction="c"
+guicmd="rofi -dmenu -i"
```

Comment:
Please add the dmenu lines back, people using dmenu might also use the script
```suggestion
guicmd="rofi -dmenu -i" #uncomment next line for dmenu
#guicmd="dmenu -i -l 15"
```

Reply:
Sorry, I forgot to add it back.